lib/phoenix/live_dashboard/helpers/tree_helpers/reingold_tifford_algorithm.ex

```diff
@@ -0,0 +1,239 @@
+    top = search_contour({node, %{}, 1, 0}, :top)
+    bottom = search_contour({top_most_sibling, %{}, 1, 0}, :bottom)
+
+    distance =
+      [Map.values(top), Map.values(bottom)]
+      |> Enum.zip()
+      |> Enum.reduce(0, fn {t, b}, acc ->
+        if t - b + acc < @total_y_distance do
+          @total_y_distance - (t - b)
+        else
+          acc
+        end
+      end)
+
+    if distance > 0 do
+      new_node = %{
+        node
+        | y: node.y + distance,
+          modifier: node.modifier + distance
+      }
+
+      fix_sibling_conflicts(new_node, other_siblings)
+    else
+      fix_sibling_conflicts(node, other_siblings)
+    end
+  end
+
+  def fix_sibling_conflicts(node, []), do: node
+
+  def search_contour({node, contour, level, mod_sum}, :top) do
+    result =
+      if Map.has_key?(contour, level) do
+        Map.put(contour, level, min(contour[level], node.y + mod_sum))
+      else
+        Map.put(contour, level, node.y + mod_sum)
+      end
+
+    Enum.reduce(
+      node.children,
+      result,
+      &search_contour({&1, &2, level + 1, mod_sum + node.modifier}, :top)
+    )
+  end
+
+  def search_contour({node, contour, level, mod_sum}, :bottom) do
+    result =
+      if Map.has_key?(contour, level) do
+        Map.put(contour, level, max(contour[level], node.y + mod_sum))
+      else
+        Map.put(contour, level, node.y + mod_sum)
+      end
+
+    Enum.reduce(
+      node.children,
+      result,
+      &search_contour({&1, &2, level + 1, mod_sum + node.modifier}, :bottom)
+    )
+  end
+
+  defp ensure_children_inside_screen(node) do
+    result =
+      {node, %{}, 1, 0}
+      |> search_contour(:top)
+      |> Enum.reduce(0, fn {_, value}, acc ->
+        if value + acc < 0, do: value * -1, else: acc
+      end)
+
+    %{node | y: node.y + result, modifier: node.modifier + result}
+  end
+
+  defp put_x_position(%{children: children} = node, position, max_width) do
+    children =
+      Enum.reduce(
+        children,
+        [],
+        &[
+          put_x_position(&1, max_width[node.level] + position + @node_x_separation, max_width)
+          | &2
+        ]
+      )
+
+    %{node | x: position, children: children}
+  end
+
+  defp put_x_position(%{children: children} = tree, :first_call) do
+    max_width = find_max_width_by_level(tree, %{}) |> IO.inspect()
+
+    children =
+      Enum.reduce(
+        children,
+        [],
+        &[put_x_position(&1, tree.width + @node_x_separation, max_width) | &2]
+      )
+
+    %{tree | x: 0, children: children}
+  end
+
+  defp find_max_width_by_level(node, max_values) do
+    max_values =
+      if Map.has_key?(max_values, node.level) do
+        Map.put(max_values, node.level, max(max_values[node.level], node.width))
+      else
+        Map.put(max_values, node.level, node.width)
+      end
+
+    Enum.reduce(
+      node.children,
+      max_values,
+      &find_max_width_by_level(&1, &2)
+    )
+  end
+
+  defp name({name, pid}) do
```

Comment:
As mentioned, we would move this out of this module, as it shouldn't be the concern of this module.